### PR TITLE
feat: Custom layouts should be hidden from sitemap

### PR DESCRIPTION
**Pull request by UttamSharma007**
- Custom layouts should be hidden from sitemap

### CODE REVIEW

File: inc/compatibility/class-astra-yoast-seo.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * Contact Form 7 Compatibility File.
```
Comment (premanshup) [inline inc/compatibility/class-astra-yoast-seo.php]: Please change comment for `Contact Form 7 Compatibility File` to Yoast SEO.

File: inc/compatibility/class-astra-yoast-seo.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ *
+ * @package Astra
+ */
+
+/**
+ * Astra Contact Form 7 Compatibility
```
Comment (premanshup) [inline inc/compatibility/class-astra-yoast-seo.php]: Please change comment for `Astra Contact Form 7 Compatibility` to Yoast SEO.

File: inc/compatibility/class-astra-yoast-seo.php [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+ */
+if ( ! class_exists( 'Astra_Yoast_SEO' ) ) :
+
+	/**
+	 * Astra Contact Form 7 Compatibility
```
Comment (premanshup) [inline inc/compatibility/class-astra-yoast-seo.php]: Please change comment for `Astra Contact Form 7 Compatibility` to Yoast SEO.

File: inc/compatibility/class-astra-yoast-seo.php [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+	 *
+	 * @since 1.0.0
```
Comment (premanshup) [inline inc/compatibility/class-astra-yoast-seo.php]: Should be `x.x.x`

File: inc/compatibility/class-astra-yoast-seo.php [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+		public function __construct() {
+			add_filter( 'wpseo_sitemap_exclude_post_type', array( $this, 'ast_sitemap_exclude_post_type' ), 10, 2 );
+		}
+
+		/**
+		 * Exclude One Content Type From Yoast SEO Sitemap
+		 *
+		 * @param  string $value value.
+		 * @param  string $post_type Post Type.
+		 * @since 1.0.0
```
Comment (premanshup) [inline inc/compatibility/class-astra-yoast-seo.php]: Should be `x.x.x`

File: inc/compatibility/class-astra-yoast-seo.php [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+		}
+
+	}
+
+endif;
+
+/**
+ * Kicking this off by calling 'get_instance()' method
+ */
+Astra_Yoast_SEO::get_instance();
```
Comment (premanshup) [inline inc/compatibility/class-astra-yoast-seo.php]: We can replace `Astra_Yoast_SEO::get_instance();` with just object creation `new Astra_Yoast_SEO();` and remove 

```
/**
		 * Member Variable
		 *
		 * @var object instance
		 */
		private static $instance;
		/**
		 * Initiator
		 */
		public static function get_instance() {
			if ( ! isset( self::$instance ) ) {
				self::$instance = new self();
			}
			return self::$instance;
		}
```

File: inc/compatibility/class-astra-yoast-seo.php [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+/**
+ * Contact Form 7 Compatibility File.
+ *
+ * @package Astra
+ */
+
+/**
+ * Astra Contact Form 7 Compatibility
+ */
+if ( ! class_exists( 'Astra_Yoast_SEO' ) ) :
```
Comment (premanshup) [inline inc/compatibility/class-astra-yoast-seo.php]: We can remove `if ( ! class_exists( 'Astra_Yoast_SEO' ) ) :` class exists check.

File: inc/compatibility/class-astra-yoast-seo.php [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+				self::$instance = new self();
+			}
+			return self::$instance;
+		}
+
+		/**
+		 * Constructor
+		 */
+		public function __construct() {
+			add_filter( 'wpseo_sitemap_exclude_post_type', array( $this, 'ast_sitemap_exclude_post_type' ), 10, 2 );
```
Comment (premanshup) [inline inc/compatibility/class-astra-yoast-seo.php]: `ast_sitemap_exclude_post_type` -> `sitemap_exclude_post_type`

File: inc/compatibility/class-astra-yoast-seo.php [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+		}
+
+		/**
+		 * Exclude One Content Type From Yoast SEO Sitemap
+		 *
+		 * @param  string $value value.
+		 * @param  string $post_type Post Type.
+		 * @since 1.0.0
+		 */
+		function ast_sitemap_exclude_post_type( $value, $post_type ) {
```
Comment (premanshup) [inline inc/compatibility/class-astra-yoast-seo.php]: `ast_sitemap_exclude_post_type` -> `sitemap_exclude_post_type`

File: inc/compatibility/class-astra-yoast-seo.php [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+			if ( 'astra-advanced-hook' == $post_type ) {
```
Comment (premanshup) [inline inc/compatibility/class-astra-yoast-seo.php]: Please do strict type checking.

File: inc/compatibility/class-astra-yoast-seo.php (hunk 1 above)
Comment (UttamSharma007) [reply inc/compatibility/class-astra-yoast-seo.php]: > Please change comment for `Contact Form 7 Compatibility File` to Yoast SEO.

Updated!

File: inc/compatibility/class-astra-yoast-seo.php (hunk 2 above)
Comment (UttamSharma007) [reply inc/compatibility/class-astra-yoast-seo.php]: > Please change comment for `Astra Contact Form 7 Compatibility` to Yoast SEO.

Updated!

File: inc/compatibility/class-astra-yoast-seo.php (hunk 3 above)
Comment (UttamSharma007) [reply inc/compatibility/class-astra-yoast-seo.php]: Updated!

File: inc/compatibility/class-astra-yoast-seo.php (hunk 4 above)
Comment (UttamSharma007) [reply inc/compatibility/class-astra-yoast-seo.php]: Updated!

File: inc/compatibility/class-astra-yoast-seo.php (hunk 8 above)
Comment (UttamSharma007) [reply inc/compatibility/class-astra-yoast-seo.php]: Updated!

File: inc/compatibility/class-astra-yoast-seo.php (hunk 9 above)
Comment (UttamSharma007) [reply inc/compatibility/class-astra-yoast-seo.php]: Updated!

File: inc/compatibility/class-astra-yoast-seo.php (hunk 10 above)
Comment (UttamSharma007) [reply inc/compatibility/class-astra-yoast-seo.php]: Done!

File: inc/compatibility/class-astra-yoast-seo.php (hunk 6 above)
Comment (UttamSharma007) [reply inc/compatibility/class-astra-yoast-seo.php]: Done!

File: inc/compatibility/class-astra-yoast-seo.php (hunk 7 above)
Comment (UttamSharma007) [reply inc/compatibility/class-astra-yoast-seo.php]: I have followed other compatible classes as a reference, so I think it fine.

Comment (Nikschavan) [reply inc/compatibility/class-astra-yoast-seo.php]: @UttamSharma007 There actually is no need to make these functions pluggable. See #783 for more explanation regarding this in the context of a function.

Comment (UttamSharma007) [reply inc/compatibility/class-astra-yoast-seo.php]: removed!

**Pull request merged**